redsys/__init__.py

```diff
@@ -133,6 +134,16 @@ def redsys_generate_request(self, params):
                 params.get('DS_MERCHANT_CONSUMERLANGUAGE'), '001'),
             }
 
+        # This field is only needed when we use some special, payment, if we
+        # dont use any, we need to remove it.
+        # Othewise the payment will fail with error: SIS0231
+        if self.paymethod == 'bizum':
+            merchant_parameters['DS_MERCHANT_PAYMETHODS'] = 'z'
+        elif self.paymethod == 'google':
```

Comment:
google and apple use same DS_MERCHANT_PAYMETHODS = xpay?

Reply:
Putting "google" in the "DS_MERCHANT_PAYMETHODS" parameters, we allow only "Google Pay" option. As documentation says:
"_Si se quiere forzar la ejecución del método de pago Google Pay (en lugar de mostrar la página de introducción de datos de tarjeta), se debe enviar en el campo DS_MERCHANT_PAYMETHODS el valor ‘google’ (sin comillas)._"
More info here: https://pagosonline.redsys.es/googlePay.html

With "xpay" we can use both methods at the same time (if the methods are configured). As documentation says:
"_En caso de tener configurado tambien Apple Pay y el dispositivo lo permite, si se quiere forzar la ejecución solamente de uno de estos método de pago se debe enviar en el campo DS_MERCHANT_PAYMETHODS el valor ‘xpay’ (sin comillas). Y después el titular pulsará sobre Google Pay o Apple Pay._"
More info here: https://pagosonline.redsys.es/applePay.html

Reply:
Ok. I think is mistake option in case google use "xpay" in https://pagosonline.redsys.es/parametros-entrada-salida.html